### PR TITLE
Holy water bead code framework

### PR DESCRIPTION
**Pull request by EltonZhang777**
Adds RB504 bead code framework
Actual beam code still pending.

### CODE REVIEW

File: Blasphemous.LostDreams/Effects/HolyWater.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using HarmonyLib;
+using UnityEngine;
+
```
Comment (BrandenEK) [inline Blasphemous.LostDreams/Effects/HolyWater.cs:7]: What is going on with the using statements up here?

**Review by BrandenEK — CHANGES_REQUESTED** [review]
Overall the code looks like a good start, but you might have to redo these commits since there will be merge conflicts because this bead was supposed to be RB506.  Did you not base your branch off of my 'health-beams' branch, since if you did you should have seen the commit and changes for RB506 stuff?

File: Blasphemous.LostDreams/Effects/HolyWater.cs (hunk 1 above)
Comment (EltonZhang777) [reply Blasphemous.LostDreams/Effects/HolyWater.cs:7]: My VS2022 automatically adds a lot of using statements automatically when I call for random stuff. I'll develop a habit to delete those unused `using` statements after finishing code

**Comment by EltonZhang777** [comment]
I'll move these code to the new branch, while experimenting a way to actually register a mod attack. 

I hope I can figure out how to create a modded attack, I'm getting close to it

**Pull request closed without merging**